src/serialization/serialize-assets.js

```diff
@@ -16,12 +16,10 @@ const serializeAssets = function (runtime, assetType, optTargetId) {
         const currAssets = currTarget.sprite[assetType];
         for (let j = 0; j < currAssets.length; j++) {
             const currAsset = currAssets[j];
-            const assetId = currAsset.assetId;
-            const storage = runtime.storage;
-            const storedAsset = storage.get(assetId);
+            const asset = currAsset.asset;
             assetDescs.push({
-                fileName: `${assetId}.${storedAsset.dataFormat}`,
-                fileContent: storedAsset.data});
+                fileName: `${asset.assetId}.${asset.dataFormat}`,
```

Comment:
It seems like there's a lot of duplicate info flying around now...

e.g. both the costume/sound object and the assets themselves have assetId (and dataFormat?) now...

We should probably clean these at some point, but that's definitely a change for the future as I'm sure there's a lot of references to the properties on the top-level objects right now.

Reply:
Agreed — I sort of touched on that in the previous comment. There should be a second phase where we re-adjust everything to be better aware of the fact that there's an asset object sitting there that might have everything you need, so you don't need to have md5, assetId, etc. sitting on the top-level object at the same time.